compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -295,6 +295,7 @@ pub(super) fn transcribe<'a>(
                         // Emit as a token stream within `Delimiter::Invisible` to maintain
                         // parsing priorities.
                         marker.visit_span(&mut sp);
+                        with_metavar_spans(|mspans| mspans.insert(mk_span, sp));
```

Comment:
This change could have gone in with #133436. It's to account for recent changes in #134478. None of our tests needed this `with_metavar_span` call for `NtVis`/`NtTy`, but we do have tests that need it for `NtPat` and `NtExpr`.